controller/network/link_controller.go

```diff
@@ -31,18 +32,65 @@ type linkController struct {
 	idGenerator    idgen.Generator
 	lock           sync.Mutex
 	initialLatency time.Duration
+	store          *objectz.ObjectStore[*Link]
 }
 
 func newLinkController(options *Options) *linkController {
 	initialLatency := DefaultOptionsInitialLinkLatency
 	if options != nil {
 		initialLatency = options.InitialLinkLatency
 	}
-	return &linkController{
+
+	result := &linkController{
 		linkTable:      newLinkTable(),
 		idGenerator:    idgen.NewGenerator(),
 		initialLatency: initialLatency,
 	}
+
+	result.store = objectz.NewObjectStore[*Link](func() objectz.ObjectIterator[*Link] {
+		return IterateCMap[*Link](result.linkTable.links)
+	})
+
+	result.store.AddStringSymbol("id", func(entity *Link) *string {
```

Comment:
It seems odd that these are outside of the store creation. Does this type of store have multiple instances?

Reply:
Yes, there isn't a store type per entity here. The store is generic, it just needs symbols and an entity iterator.